File: 2019-11-11-381-easy-yahtzee-upper-section-scoring/main.py
def yahtzee_upper(dicerolls):
    return max(i * dicerolls.count(i) for i in range(1, 7))

print(yahtzee_upper([2, 3, 5, 5, 6])) # => 10
print(yahtzee_upper([1, 1, 1, 1, 3])) # => 4
print(yahtzee_upper([1, 1, 1, 3, 3])) # => 6
print(yahtzee_upper([1, 2, 3, 4, 5])) # => 5
print(yahtzee_upper([6, 6, 6, 6, 6])) # => 30

File: 2019-11-11-381-easy-yahtzee-upper-section-scoring/main.swift
import Foundation

func yahtzeeUpper(_ dicerolls: [Int]) -> Int {
    var ret = 0
    for n in 1...6 {
        let score = dicerolls.filter { $0 == n }.count * n
        ret = max(ret, score)
    }
    return ret
}

print(yahtzeeUpper([2, 3, 5, 5, 6])) // => 10
print(yahtzeeUpper([1, 1, 1, 1, 3])) // => 4
print(yahtzeeUpper([1, 1, 1, 3, 3])) // => 6
print(yahtzeeUpper([1, 2, 3, 4, 5])) // => 5
print(yahtzeeUpper([6, 6, 6, 6, 6])) // => 30